test/SideNav.spec.js

```diff
@@ -26,71 +26,92 @@ function setup(props = {}, children, mounted) {
 }
 
 describe('<SideNav />', () => {
-  const sideNavMock = jest.fn();
-  const restore = mocker('sidenav', sideNavMock);
+  describe('initialises', () => {
+    const sidenavInitMock = jest.fn();
+    const sidenavInstanceDestroyMock = jest.fn();
+    const sidenavMock = {
+      init: (el, options) => {
+        sidenavInitMock(options);
+        return {
+          destroy: sidenavInstanceDestroyMock
+        };
+      }
+    };
+    const restore = mocker('Sidenav', sidenavMock);
 
-  afterAll(() => {
-    restore();
-  });
+    beforeEach(() => {
+      sidenavInitMock.mockClear();
+      sidenavInstanceDestroyMock.mockClear();
+    });
 
-  test('renders', () => {
-    const { sideNav } = setup();
-    expect(sideNav).toMatchSnapshot();
-  });
+    afterAll(() => {
+      restore();
+    });
 
-  test('can be `fixed`', () => {
-    const { sideNav } = setup({ className: 'red', fixed: true });
-    expect(sideNav).toMatchSnapshot();
-  });
+    test('calls SideNav', () => {
+      mount(<SideNav />);
+      expect(sidenavInitMock).toHaveBeenCalledTimes(1);
+    });
 
-  test('should render fixed if not passed trigger or fixed', () => {
-    const { wrapper } = setup({ className: 'red' });
-    expect(wrapper).toMatchSnapshot();
-  });
+    test('renders', () => {
+      const { sideNav } = setup();
+      expect(sideNav).toMatchSnapshot();
+    });
 
-  test('should render trigger on large screens not fixed`', () => {
-    const { trigger } = setup({ className: 'green' });
-    expect(trigger).toMatchSnapshot();
-  });
+    test('can be `fixed`', () => {
+      const { sideNav } = setup({ className: 'red', fixed: true });
+      expect(sideNav).toMatchSnapshot();
+    });
 
-  test('should be fixed if no trigger is passed', () => {
-    const component = <SideNav className="red" />;
-    const wrapper = shallow(component);
-    const sideNav = wrapper.find('.sidenav');
-    expect(sideNav).toMatchSnapshot();
-  });
+    test('should render fixed if not passed trigger or fixed', () => {
+      const { wrapper } = setup({ className: 'red' });
+      expect(wrapper).toMatchSnapshot();
+    });
 
-  test('should render a given id', () => {
-    const { sideNavProps, triggerProps } = setup({ id: 'test' });
-    expect(sideNavProps.id).toEqual('test');
-    expect(triggerProps['data-target']).toEqual('test');
-  });
+    test('should render trigger on large screens not fixed`', () => {
+      const { trigger } = setup({ className: 'green' });
+      expect(trigger).toMatchSnapshot();
+    });
```

Comment:
is this test useful? how to check it's a large screen?

Reply:
Dropped it